threads/main.go

```diff
@@ -0,0 +1,210 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/c-bata/go-prompt"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/textileio/go-threads/api/client"
+	"github.com/textileio/go-threads/util"
+	"google.golang.org/grpc"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+
+var (
+	apiClient      *client.Client
+	apiTimeout     = time.Second * 5
+	streamTimeout  = time.Second * 30
+	noStoreMessage = "Store required. `use <store id>`"
+	currentStore   string
+	suggestions    = []prompt.Suggest{
+		{
+			Text:        "use",
+			Description: "Switch the store.",
+		},
+		{
+			Text:        "exit",
+			Description: "Exit.",
+		},
+	}
+	storeSuggestions = []prompt.Suggest{
+		{
+			Text:        "listen",
+			Description: "Stream all store updates.",
+		},
+		{
+			Text:        "getModel",
+			Description: "Get all models by model name.",
+		},
+	}
+)
+
+type any map[string]interface{}
+
+func completer(in prompt.Document) []prompt.Suggest {
+	w := in.GetWordBeforeCursor()
+	if w == "" {
+		return []prompt.Suggest{}
+	}
+	if currentStore != "" {
+		return prompt.FilterHasPrefix(append(suggestions, storeSuggestions...), w, true)
+	}
+	return prompt.FilterHasPrefix(suggestions, w, true)
+}
+
+func storeExecutor(blocks []string) {
+	switch blocks[0] {
+	case "listen":
+		listen(currentStore)
+		return
+	case "getModel":
+		if len(blocks) < 2 {
+			fmt.Println("You must provide a model name.")
+			return
+		}
+		getModel(currentStore, blocks[1], apiTimeout)
+		return
+	default:
+		fmt.Println("Sorry, I don't understand.")
+	}
+}
+
+func executor(in string) {
+	in = strings.TrimSpace(in)
+	blocks := strings.Split(in, " ")
+	switch blocks[0] {
+	case "":
+		return
+	case "use":
+		if len(blocks) < 2 {
+			fmt.Println("You must provide a store ID.")
+			return
+		}
+		currentStore = blocks[1]
+		fmt.Printf("Switched to %s\n", blocks[1])
+		return
+	case "exit":
+		fmt.Println("Bye!")
+		os.Exit(0)
+	default:
+		if currentStore == "" {
+			fmt.Println(noStoreMessage)
+			return
+		}
+		storeExecutor(blocks)
+	}
+}
+
+func getModel(id string, model string, maxAwaitTime time.Duration) {
+	ctx, cancel := context.WithTimeout(context.Background(), apiTimeout)
+	defer cancel()
+
+	rawResults, err := apiClient.ModelFind(ctx, id, model, nil, []*any{})
```

Comment:
`nil` here hurts readability a little, IMO... you could just pass an empty query: `JSONQuery{}`. In fact, for this reason, this method should probably not take a pointer.